graylog2-server/src/main/java/org/graylog2/configuration/HttpConfiguration.java

```diff
@@ -0,0 +1,253 @@
+import com.github.joschi.jadconfig.validators.URIAbsoluteValidator;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.net.HostAndPort;
+import com.google.common.net.InetAddresses;
+import org.graylog2.plugin.Tools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class HttpConfiguration {
+    private static final Logger LOG = LoggerFactory.getLogger(HttpConfiguration.class);
+
+    private static final String WILDCARD_IP_ADDRESS = "0.0.0.0";
+    private static final int GRAYLOG_DEFAULT_PORT = 9000;
+
+    public static final String OVERRIDE_HEADER = "X-Graylog-Server-URL";
+    public static final String PATH_WEB = "";
+    public static final String PATH_API = "api/";
+
+    @Parameter(value = "http_bind_address", required = true)
+    private HostAndPort httpBindAddress = HostAndPort.fromParts("127.0.0.1", GRAYLOG_DEFAULT_PORT);
+
+    @Parameter(value = "http_publish_uri", validator = URIAbsoluteValidator.class)
+    private URI httpPublishUri;
+
+    @Parameter(value = "http_enable_cors")
+    private boolean httpEnableCors = true;
+
+    @Parameter(value = "http_enable_gzip")
+    private boolean httpEnableGzip = true;
+
+    @Parameter(value = "http_max_header_size", required = true, validator = PositiveIntegerValidator.class)
+    private int httpMaxHeaderSize = 8192;
+
+    @Parameter(value = "http_thread_pool_size", required = true, validator = PositiveIntegerValidator.class)
+    private int httpThreadPoolSize = 16;
+
+    @Parameter(value = "http_selector_runners_count", required = true, validator = PositiveIntegerValidator.class)
+    private int httpSelectorRunnersCount = 1;
+
+    @Parameter(value = "http_enable_tls")
+    private boolean httpEnableTls = false;
+
+    @Parameter(value = "http_tls_cert_file")
+    private Path httpTlsCertFile;
+
+    @Parameter(value = "http_tls_key_file")
+    private Path httpTlsKeyFile;
+
+    @Parameter(value = "http_tls_key_password")
+    private String httpTlsKeyPassword;
+
+    @Parameter(value = "http_external_uri")
+    private URI httpExternalUri;
+
+    public HostAndPort getHttpBindAddress() {
+        return httpBindAddress
+                .requireBracketsForIPv6()
+                .withDefaultPort(9000);
+    }
+
+    public String getUriScheme() {
+        return getUriScheme(isHttpEnableTls());
+    }
+
+    public String getUriScheme(boolean enableTls) {
+        return enableTls ? "https" : "http";
+    }
+
+    public URI getHttpPublishUri() {
+        if (httpPublishUri == null) {
+            final URI defaultHttpUri = getDefaultHttpUri();
+            LOG.debug("No \"http_publish_uri\" set. Using default <{}>.", defaultHttpUri);
+            return defaultHttpUri;
+        } else if (WILDCARD_IP_ADDRESS.equals(httpPublishUri.getHost())) {
```

Comment:
This does not take the IPv6 wildcard address into account: `[::]`. See below for another instance of that.